- The long process, simulated with function "longRunningProcess" can be cancelled using the context that is provided. This is a placeholder that assumes the function is developed in such a way as to be context-cancellable. As it is, there's no way of interrupting a function that just sleeps so it could be possible that, if you can't cancel this process using context then the thread could continue running and complete the request but return an error saying that the request was cancelled. This would mean that the request caller could be configured to retry the request on a different server and cause a double-process. As long as the request is idempotent this shouldn't be a problem as the likelihood of this occurring is minimal and would therefore place negligible additional strain on the network. It would only happen if the request were to complete in the time between the requestHandler returning the error and the server shutting down.